packages/lesswrong/components/posts/AddToCalendar/AddToCalendarButton.tsx

```diff
@@ -0,0 +1,162 @@
+    <path d="M 226.571 27.045 L 226.571 29.045 L 223.571 29.045 L 223.571 32.045 L 221.571 32.045 L 221.571 29.045 L 218.571 29.045 L 218.571 27.045 L 221.571 27.045 L 221.571 24.045 L 223.571 24.045 L 223.571 27.045 L 226.571 27.045 Z"/>
+  </svg>
+}
+
+const AddToCalendarButton = ({post, label, hideTooltip, hideIcon, iconClassName, classes}: {
+  post: PostsList|PostsWithNavigation|PostsWithNavigationAndRevision,
+  label?: string,
+  hideTooltip?: boolean,
+  hideIcon?: boolean,
+  iconClassName?: string,
+  classes: ClassesType,
+}) => {
+  const { captureEvent } = useTracking()
+  const [open, setOpen] = useState(false)
+  
+  // close the dropdown when clicking on the page
+  useEffect(() => {
+    if (open) {
+      const onClose = () => setOpen(false)
+      document.addEventListener("click", onClose)
+      return () => document.removeEventListener("click", onClose)
+    }
+  }, [open, setOpen])
+  
+  const handleClick = (e) => {
+    e.preventDefault()
+    e.stopPropagation()
+    
+    if (!open) {
+      captureEvent('addToCalendarClicked')
+    }
+    setOpen(!open)
+  }
+  
+  // we use the Facebook link as the default event details text
+  let eventDetails = post.facebookLink;
+  // we try to use plaintextDescription instead if possible
+  // (only PostsList should be missing the plaintextDescription, so we pull that in)
+  const { document: data } = useSingle({
+    collectionName: "Posts",
+    fragmentName: 'PostsPlaintextDescription',
+    documentId: post._id,
+    skip: !post.startTime || !post.contents || ('plaintextDescription' in post.contents)
+  });
+  
+  if (!post.startTime) {
+    return null;
+  }
+  
+  if (data) {
+    eventDetails = data.contents?.plaintextDescription || eventDetails;
+  } else if (post.contents && 'plaintextDescription' in post.contents) {
+    eventDetails = post.contents.plaintextDescription || eventDetails;
+  }
+  
+  const endTime = post.endTime ? moment(post.endTime) : moment(post.startTime).add(1, 'hour')
+  
+  // build the links for different calendars
+  const urls = makeUrls({
+    name: post.title,
+    details: eventDetails,
+    location: post.onlineEvent ? post.joinEventLink : post.location,
+    startsAt: moment(post.startTime).format(),
+    endsAt: endTime.format()
+  })
+  
+  const calendarIconNode = (
+    <div className={classes.root}>
+      <button className={classes.button} onClick={handleClick}>
+        {!hideIcon && <AddToCalendarIcon className={classNames(classes.icon, iconClassName)} />}
+        {label && (
+          <span className={classes.label}>
+            {label}
+          </span>
+        )}
+      </button>
+      {open && <div className={classes.dropdown}>
```

Comment:
Another bonus, I got to reorder the dropdown options based on usage (so Google is first)